roles/host-storage/tasks/main.yml

```diff
@@ -16,52 +16,33 @@
 - name: Partition Oracle user mount devices
   parted:
     device: "{{ item.blk_device }}"
-    number: 1
+    number: "{{ item.num }}"
+    part_start: "{{ item.start }}"
+    part_end: "{{ item.end }}"
     state: present
     label: gpt
-  when: "'mapper' not in item.blk_device"
+  when: "ansible_hostname==item.host_name"
   with_items:
     - "{{ oracle_user_data_mounts }}"
   tags: ora-mounts
 
 - name: Add filesystem on Oracle user mount raw devices
   filesystem:
     fstype: "{{ item.fstype }}"
-    dev: "{{ item.blk_device }}1"
-  when: "'mapper' not in item.blk_device"
-  with_items:
-    - "{{ oracle_user_data_mounts }}"
-  tags: ora-mounts
-
-- name: Add filesystem on Oracle user mount LV devices
-  filesystem:
-    fstype: "{{ item.fstype }}"
-    dev: "{{ item.blk_device }}"
-  when: "'mapper' in item.blk_device"
+    dev: "{{ item.blk_device }}{{ '' if ansible_distribution == 'OracleLinux' else 'p' }}{{ item.num }}"
```

Comment:
Greetings !

Trying to understand this PR in detail.
For `{{ item.blk_device }}` can you please provide some examples of such device names and configs (additional details) where it doesn't contain `mapper` and the corresponding storage layout in the BMS host?

Maybe output from commands such as lsblk, fdisk / parted, multipath, sanlun lun show, pvs, vgs, lvs could paint a picture of the layout. (Of course assuming, you will be scrambling out the WWIDs and other such details to sanitize!)

Also, the data_mounts_config.json that was fed into `install-oracle.sh` will be helpful as well.

(I see changes to that json file proposed in the other PR #61, but it will be good to have the file that worked with the changes to host-storage/tasks/main.yml file)

Thanks

Reply:
Hello JC,

The main point of this change is allowing the toolkit to create and configure /u01 and /u02 as local mount points on RAC when every machine has its own local LUN and hence we need a way to assign LUN WWIDs to server names. Please check section "4- Edit Ansible Configuration json Files" in the following document to clarify all possible scenarios that could be achieved by modifying data_mounts_config.json
https://docs.google.com/document/d/1MRwg0kUWhTxn62GQ0-xD7HRCqVgDZJiABp4qIc_kki8/edit?usp=sharing

Reply:
Hello Ahmad,
Thanks for the doc and all your inputs on this PR #62 and PR #61.

We have identified and tested the solution for a variety of similar scenarios for local host-storage including udev-ising and kpartx-ing the devices so that any local host device presented is utilized correctly for mountpoints like, say `/u01`.

While are working on the design, coding, testing of the scenarios with you in the loop, you may pls close the loop on this PR #62 and PR #61.

Thanks